### PR TITLE
[wasm] Port use of `dirent` for wasi-libc

### PR DESCRIPTION
**Pull request by kateinoigakukun**
wasi-libc defines `d_name` as "flexible array member" which is not supported by ClangImporter yet. This patch replaces the use of `d_name` with a helper function `_direntName` to work around the issue.

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
This doesn't seem to be exclusive to WASI `#if` branches, so I'll defer to @parkera.

**Review by MaxDesiatov — APPROVED** [review]
On a second look this looks like a safe enough change, especially to unblock ongoing WASI support.

**Pull request merged**
